isis/src/voyager/apps/voy2isis/main.cpp

```diff
@@ -484,16 +490,30 @@ QByteArray fixLabels(QString fileName, History *hist){
 
   // Check if the instrument name is valid
   if (labels.contains(QByteArray("INSTRUMENT_NAME\n"))) {
-    labels.replace("INSTRUMENT_NAME", "INSTRUMENT_NAME                  = Unknown");
-
-    PvlGroup insNameWarning("Warning");
-    PvlKeyword insNameMsg("Message", "The INSTRUMENT_NAME for [" + fileName + "] is empty."
-                              + "The InstrumentId in the output cube will instead be set to "
-                              + "[Unknown] and the labels will not translate.");
-    insNameWarning += insNameMsg;
-    Application::Log(insNameWarning);
-
-    hEntry += insNameWarning;
+ 
+    // Only set name to Unknown and warn the user about a missing INSTRUMENT_NAME 
+    // if it wasn't specified by the user.
+    UserInterface &ui = Application::GetUserInterface();
+    if (ui.GetString("INSTRUMENT") == "AUTOMATIC") {
+      labels.replace("INSTRUMENT_NAME", "INSTRUMENT_NAME                  = Unknown");
+      PvlGroup insNameWarning("Warning"); 
+      PvlKeyword insNameMsg("Message", "The INSTRUMENT_NAME for [" + fileName + "] is empty."
+                                + "The InstrumentId in the output cube will instead be set to "
+                                + "[Unknown] and the labels will not translate. To create a cube "  
+                                + "with translated labels, re-run this "
+                                + "application with INSTRUMENT set to NAC or WAC.");
+
```

Comment:
Should this be an error and exit?

Reply:
I think raising an exception is a good idea because you cannot do further processing. If users want to just see the raw image data they can use pds2isis.

Reply:
@jessemapel `pds2isis` actually possibly won't work in the cases where this message is created to be displayed later. The method this code is in fixes a malformed pds label so that it can be ingested to ISIS without failing. I went ahead and tried this on the test data associated with this ticket and `pds2isis` does indeed fail with: `**I/O ERROR** Unknown label type in [c2061155.imq]. It is possible the label file does not describe an image product (IMAGE, CUBE, or SPECTRALCUBE).` I didn't add this error message or the particular way of saving it to display but not actually throwing an exception and I suspect being able to successfully create a cube with DNs but not the labels translated is probably why. @Kelvinrr It will error and exit later if it still can't find an `INSTRUMENT_NAME` to use.

Reply:
It still seems best to error out sooner than later. Even if you just wanted raw image data, getting a hard error and then re-running with a forced instrument name seems reasonable enough?

Reply:
@Kelvinrr I'm reluctant to change the way that `voy2isis` currently functions on images with broken input labels unless that's also a desired change from the user end? The relevant change in behavior would be that no output cube would be created if an error in parsing the label was encountered, and the program would begin to exit with a non-zero exit code, both of which could break scripts that currently use this application. There are more ways that parsing the pds label can fail than just lacking an `INSTRUMENT_NAME`, for example the second half of that function deals with a potentially broken `IMAGE_ID` and it seems odd to handle them inconsistently. (We did just add the option to specify one by-hand, but not the other, _but_ the `IMAGE_ID` can be populated with an alternative value without problem.)

The strategy of "ingest the image with as much of the label as possible" was added as part of https://fixit.wr.usgs.gov/fixit/issues/4421 (not migrated to github?) which has substantial discussion I'm still looking through on it. Looking at the code, it seems to me that in the past, someone made a deliberate effort to not have the program crash if the label was malformed, and if we're going to modify this behavior I'd like to understand why before doing so.

Reply:
It sounds like Tammy was pulling for doing as much as possible. In that case I'm okay with sticking with that.